From 61c94f9b499d1621021c82ba7149d46ba4d6e9f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Old=C5=99ich=20Jedli=C4=8Dka?= <oldium.pro@gmail.com>
Date: Sat, 28 Sep 2024 22:47:04 +0200
Subject: [PATCH] pkcs11: add Dracut install check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The check allows installation of all Dracut files together, as it is done
with Debian packaging, so that it works even without having pkcs11 pin
installed.

Signed-off-by: Oldřich Jedlička <oldium.pro@gmail.com>
---
 src/luks/dracut/clevis-pin-pkcs11/module-setup.sh.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/luks/dracut/clevis-pin-pkcs11/module-setup.sh.in b/src/luks/dracut/clevis-pin-pkcs11/module-setup.sh.in
index 39d06a04..4e757d67 100755
--- a/src/luks/dracut/clevis-pin-pkcs11/module-setup.sh.in
+++ b/src/luks/dracut/clevis-pin-pkcs11/module-setup.sh.in
@@ -17,6 +17,13 @@
 #
 # shellcheck disable=SC2154
 #
+
+check() {
+    require_binaries pcscd pkcs11-tool clevis-decrypt-pkcs11 || return 1
+    require_binaries awk head sed socat tail tr || return 1
+    return 0
+}
+
 depends() {
     echo clevis
     return 255
